src/uct/ugni/rdma/ugni_rdma_ep.c

```diff
@@ -329,12 +330,128 @@ ucs_status_t uct_ugni_ep_atomic_cswap32(uct_ep_h tl_ep, uint32_t compare, uint32
     return UCS_ERR_UNSUPPORTED;
 }
 
```

Comment:
I'd suggest rather than a big ifdef, just detect at runtime whether or not the the app is on a gem or aries system.